app/src/main/java/com/github/se/cyrcle/ui/profile/EditProfileComponent.kt

```diff
@@ -48,17 +48,21 @@ fun EditProfileComponent(
   var firstName by remember { mutableStateOf(user.details!!.firstName) }
   var lastName by remember { mutableStateOf(user.details!!.lastName) }
   var profilePictureUri by remember { mutableStateOf(user.localSession?.profilePictureUri) }
+  val profilePictureUrl by remember { mutableStateOf(user.localSession?.profilePictureUrl) }
+  // To display the local profile picture when the user changes it, and hasn't uploaded (saved) it
+  var hasTemporarilyChangedProfilePicture by remember { mutableStateOf(false) }
 
   val imagePickerLauncher =
       rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri ->
         uri?.let { profilePictureUri = it.toString() }
+        hasTemporarilyChangedProfilePicture = true
```

Comment:
what's the use of the mutableState here? it starts at false then is statically moved to true, so I don't see its utility, probably just me though...

Reply:
I added a comment where I declare it to make it clear, but if it is not clear enough :


When you have selected the new profile picture but haven't click save yet, you want to see your profile with the new picture you just selected, but the picture hasn't been uploaded yet so you have to get it from the local storage this is where this state comes in to switch between displaying local storage or from the database

Reply:
Thanks for clarifying!